### PR TITLE
fxa-settings(storybook): add account_recovery_reset_password

### PR DESCRIPTION
**Pull request by millsoper**
## Because:

* We're making React versions of content server's remaining backbone pages, and moving them first into Storybook.

## This commit:

* Sets up the page view metric, basic tests, and storybook stories for the account_recovery_reset_password page.

Closes #https://mozilla-hub.atlassian.net/browse/FXA-6343

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)
Screen Shot 2022-12-28 at 2 38 32 PM
Screen Shot 2022-12-28 at 2 38 37 PM
Screen Shot 2022-12-28 at 2 38 43 PM

## Other information (Optional)
There are a number of remaining TO-DOs noted in the PR. They fall outside of the scope of this ticket. 
There are also some small variations from the original pages (for example, the `Remembered password?` link is now, `Remembered your password? Sign in`.) This is to match already-established precedent in the Settings app.

### CODE REVIEW

File: packages/fxa-settings/src/pages/AccountRecoveryResetPassword/en.ftl [hunk 1]
```diff
@@ -0,0 +1,8 @@
+password-link-expired = Reset password link expired
```
Comment (bcolsson) [inline packages/fxa-settings/src/pages/AccountRecoveryResetPassword/en.ftl:1]:
```suggestion
## Account recovery reset password page

#  Appears when a link to reset password has expired
password-link-expired-header = Reset password link expired
```

File: packages/fxa-settings/src/pages/AccountRecoveryResetPassword/en.ftl [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+password-link-damaged = Reset password link damaged
```
Comment (bcolsson) [inline packages/fxa-settings/src/pages/AccountRecoveryResetPassword/en.ftl:2]:
```suggestion
# Appears when a link to reset password is damaged
password-link-damaged-header = Reset password link damaged
```

File: packages/fxa-settings/src/pages/AccountRecoveryResetPassword/en.ftl [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+create-new-password = Create new password
```
Comment (bcolsson) [inline packages/fxa-settings/src/pages/AccountRecoveryResetPassword/en.ftl:3]:
```suggestion
# Header for form to create new password
create-new-password-header = Create new password
```

File: packages/fxa-settings/src/pages/AccountRecoveryResetPassword/en.ftl [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+receive-new-link = Receive new link
```
Comment (bcolsson) [inline packages/fxa-settings/src/pages/AccountRecoveryResetPassword/en.ftl:4]:
```suggestion
# Link that user can click to receive a new reset password link
receive-new-link = Receive new link
```

File: packages/fxa-settings/src/pages/AccountRecoveryResetPassword/en.ftl [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+confirm-account-recovery-key-button = Reset password
+account-restored-success = You have successfully restored your account using your account recovery key. Create a new password to secure your data, and store it in a safe location.
```
Comment (bcolsson) [inline packages/fxa-settings/src/pages/AccountRecoveryResetPassword/en.ftl:6]:
```suggestion
account-restored-success-message = You have successfully restored your account using your account recovery key. Create a new password to secure your data, and store it in a safe location.
```

File: packages/fxa-settings/src/pages/AccountRecoveryResetPassword/en.ftl (hunk 1 above)
Comment (bcolsson) [reply packages/fxa-settings/src/pages/AccountRecoveryResetPassword/en.ftl:1]: Add group level comment and individual level comment to provide context for localizers.

Suggest tweaking ID to also provide context on how this appears. (More important for something like "Create new password" below which could be confused for a button/link.)

File: packages/fxa-settings/src/pages/AccountRecoveryResetPassword/en.ftl (hunk 2 above)
Comment (bcolsson) [reply packages/fxa-settings/src/pages/AccountRecoveryResetPassword/en.ftl:2]: Individual level comment and tweak to ID to provide context for localizers. "Reset password link damaged" may be hard to parse without context.

File: packages/fxa-settings/src/pages/AccountRecoveryResetPassword/en.ftl (hunk 3 above)
Comment (bcolsson) [reply packages/fxa-settings/src/pages/AccountRecoveryResetPassword/en.ftl:3]: This could easily be confused for text for something like a button or link to create a new password, so ID and comment needed for localizers.

File: packages/fxa-settings/src/pages/AccountRecoveryResetPassword/en.ftl (hunk 4 above)
Comment (bcolsson) [reply packages/fxa-settings/src/pages/AccountRecoveryResetPassword/en.ftl:4]: Comment added to provide more context for localizers, since this is an action rather than something like header text.

File: packages/fxa-settings/src/pages/AccountRecoveryResetPassword/en.ftl (hunk 5 above)
Comment (bcolsson) [reply packages/fxa-settings/src/pages/AccountRecoveryResetPassword/en.ftl:6]: Added ID change to align with other message IDs.

**Review by bcolsson — CHANGES_REQUESTED** [review]

File: packages/fxa-settings/src/pages/AccountRecoveryResetPassword/index.test.tsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+}));
+
+const mockNavigate = jest.fn();
+jest.mock('@reach/router', () => ({
+  ...jest.requireActual('@reach/router'),
+  useNavigate: () => mockNavigate,
+}));
+
+describe('PageAccountRecoveryResetPassword', () => {
+  // TODO: enable l10n tests when they've been updated to handle embedded tags in ftl strings
```
Comment (vbudhram) [inline packages/fxa-settings/src/pages/AccountRecoveryResetPassword/index.test.tsx:25]: 👍🏽

File: packages/fxa-settings/src/pages/AccountRecoveryResetPassword/index.tsx [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+  // - focusing the password inputs if desired
+  // - remembering one's password
+  // - asking for a new link.
+  // * Hook up the functionality for sending a user a new verification link,
+  // instead of this dummy function.
+  const sendNewLinkEmail = () => {};
+
+  const onSubmit = () => {
+    try {
+      account.changePassword(currentPassword, newPassword);
```
Comment (vbudhram) [inline packages/fxa-settings/src/pages/AccountRecoveryResetPassword/index.tsx:88]: The `changePassword` assumes you are logged in already, this is where we will need to add support for new gql mutations from https://github.com/mozilla/fxa/pull/14564.

**Review by vbudhram — APPROVED** [review]
Thanks @millsoper, I don't have any issues with landing these views (other than l10n).

**Review by bcolsson — APPROVED** [review]
